src/rrulestr.ts

```diff
@@ -252,72 +347,54 @@ class RRuleStr {
     }
   }
 
-  parse (s, options) {
-    options = options || {}
-
-    const invalid = []
+  parse (s: string, options: Partial<RRuleStrOptions> = {}): RRule | RRuleSet {
+    const invalid: any[] = []
     const keys = Object.keys(options)
     const defaultKeys = Object.keys(RRuleStr.DEFAULT_OPTIONS)
 
     keys.forEach(function (key) {
       if (!contains(defaultKeys, key)) invalid.push(key)
     }, this)
 
-    if (invalid.length) throw new Error('Invalid options: ' + invalid.join(', '))
+    if (invalid.length) {
+      throw new Error('Invalid options: ' + invalid.join(', '))
+    }
 
     // Merge in default options
-    defaultKeys.forEach(function (key) {
+    defaultKeys.forEach(function (key: keyof Options) {
+      // @ts-ignore
       if (!contains(keys, key)) options[key] = RRuleStr.DEFAULT_OPTIONS[key]
     })
 
     return this._parseRfc(s, options)
   }
-}
-
-RRuleStr.DEFAULT_OPTIONS = {
-  dtstart: null,
-  cache: false,
-  unfold: false,
-  forceset: false,
-  compatible: false,
-  ignoretz: false,
-  tzinfos: null
-}
-
-RRuleStr._freq_map = {
-  'YEARLY': RRule.YEARLY,
-  'MONTHLY': RRule.MONTHLY,
-  'WEEKLY': RRule.WEEKLY,
-  'DAILY': RRule.DAILY,
-  'HOURLY': RRule.HOURLY,
-  'MINUTELY': RRule.MINUTELY,
-  'SECONDLY': RRule.SECONDLY
-}
-
-RRuleStr._weekday_map = {
-  'MO': 0,
-  'TU': 1,
-  'WE': 2,
-  'TH': 3,
-  'FR': 4,
-  'SA': 5,
-  'SU': 6
-}
 
-RRuleStr.prototype._handle_DTSTART = function (rrkwargs, name, value, options) {
-  rrkwargs[name.toLowerCase()] = dateutil.untilStringToDate(value)
+  // tslint:disable-next-line:variable-name
```

Comment:
Maybe do a `// tslint:disable:variable-name` here and `// tslint:enable:variable-name` after this block of stuff